specs/lib/gltfPipelineSpec.js

```diff
@@ -50,92 +51,111 @@ describe('gltfPipeline', function() {
+    // FAIL
     it('will write a file to the correct directory', function(done) {
-        var spy = spyOn(fsExtra, 'outputJson').and.callFake(function(file, data, callback) {
-            callback();
-        });
+        var spy = spyOn(fsExtra, 'outputJson').and.callFake(function() {});
         var options = {
             createDirectory : false
         };
-        processFileToDisk(gltfPath, outputGltfPath, options, function() {
-            expect(path.normalize(spy.calls.first().args[0])).toEqual(path.normalize(outputGltfPath));
-            done();
-        });
+        processFileToDisk(gltfPath, outputGltfPath, options)
+            .then(function() {
+                expect(path.normalize(spy.calls.first().args[0])).toEqual(path.normalize(outputGltfPath));
+                done();
+            })
+            .catch(function(err) {
+                done();
+                throw err;
+            });
     });
-
+    // FAIL
     it('will write a binary file', function(done) {
-        var spy = spyOn(fsExtra, 'outputFile').and.callFake(function(file, data, callback) {
-            callback();
-        });
+        var spy = spyOn(fsExtra, 'outputFile').and.callFake(function() {});
         var options = {
             binary : true,
             createDirectory : false
         };
-        processFileToDisk(gltfPath, outputGlbPath, options, function() {
-            expect(path.normalize(spy.calls.first().args[0])).toEqual(path.normalize(outputGlbPath));
-            done();
-        });
-    });
+        processFileToDisk(gltfPath, outputGlbPath, options)
+            .then(function() {
+                expect(path.normalize(spy.calls.first().args[0])).toEqual(path.normalize(outputGlbPath));
+                done();
+            })
+            .catch(function(err) {
+                done();
+                throw err;
+            });
 
+    });
+    // FAIL
     it('will write a file from JSON', function(done) {
-        var spy = spyOn(fsExtra, 'outputJson').and.callFake(function(file, data, callback) {
-            callback();
+        var spy = spyOn(fsExtra, 'outputJson').and.callFake(function() {
+            return;
         });
-        var options = {
-            createDirectory : false,
+        var readOptions = {};
+        var processOptions = {
+            createDirectory : false, 
             basePath : path.dirname(gltfPath)
         };
-        readGltf(gltfPath, options, function(gltf) {
-            processJSONToDisk(gltf, outputGltfPath, options, function() {
+        readGltf(gltfPath, readOptions)
+            .then(function(gltf) {
+                return processJSONToDisk(gltf, outputGltfPath, processOptions);
+            })
+            .then(function() {
                 expect(path.normalize(spy.calls.first().args[0])).toEqual(path.normalize(outputGltfPath));
                 done();
+            })
+            .catch(function(err) {
+                done();
+                throw err;
```

Comment:
The three above tests are failing

The spy is never being triggered and I am not exactly sure why. I do a similar spy setup for `writeGltfSpec` and `writeBinaryGltfSpec` that work fine. 

@lasalvavida or @lilleyse, if you have a minute could you briefly look over these tests and see if there is anything I'm obviously missing